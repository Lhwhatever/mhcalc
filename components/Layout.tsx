import React from 'react'
import Navbar from './Navbar'
import { Box, Container } from '@material-ui/core'

interface LayoutProps {
    children: React.ReactNode
}

const Layout = ({ children }: LayoutProps): JSX.Element => {
    return (
        <React.Fragment>
            <Navbar />
            <Box mt={2} />
            <Container>{children}</Container>
        </React.Fragment>
    )
}

export default Layout
