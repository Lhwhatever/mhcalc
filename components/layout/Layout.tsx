import { Box, makeStyles, Theme, Toolbar, useMediaQuery } from '@material-ui/core'
import React from 'react'
import Navbar from './Navbar'
import NavDrawer from './NavDrawer'

export const altDrawerWidth = 240
export const altDrawerMinBreakpoint = 'md'

const useStyles = makeStyles((theme) => ({
    root: { display: 'flex' },
    content: { flexGrow: 1, padding: theme.spacing(3) },
    appBar: { [theme.breakpoints.up(altDrawerMinBreakpoint)]: { zIndex: theme.zIndex.drawer + 1 } },
    drawer: { [theme.breakpoints.up(altDrawerMinBreakpoint)]: { width: altDrawerWidth, flexShrink: 0 } },
    drawerPaper: { [theme.breakpoints.up(altDrawerMinBreakpoint)]: { width: altDrawerWidth } }
}))

interface LayoutProps {
    children: React.ReactNode
}

const Layout = ({ children }: LayoutProps): JSX.Element => {
    const classes = useStyles()
    const useAltDrawer = useMediaQuery((theme: Theme) => theme.breakpoints.up(altDrawerMinBreakpoint))
    const usePersistentDrawer = useMediaQuery((theme: Theme) => theme.breakpoints.only('sm'))
    const drawerVariant = useAltDrawer ? 'permanent' : usePersistentDrawer ? 'persistent' : 'temporary'

    const [drawerOpen, setDrawerOpen] = React.useState(false)

    // Higher order function. Accepts a boolean. Returns a callback which will set the state of the drawer to the boolean.
    const createDrawerStateHandler = (drawerEndState: boolean) => (event?: React.SyntheticEvent<unknown, Event>) => {
        if (event && event.type === 'keydown') {
            const e = (event as unknown) as { key: string }
            if (e && (e.key === 'Tab' || e.key === 'Shift')) return
        }
        setDrawerOpen(drawerEndState)
    }

    const handleDrawerOpen = createDrawerStateHandler(true)
    const handleDrawerClose = createDrawerStateHandler(false)

    return (
        <div className={classes.root}>
            <Navbar className={classes.appBar} onDrawerOpen={handleDrawerOpen} showDrawerIcon={!useAltDrawer} />
            <NavDrawer
                anchor="left"
                open={drawerOpen}
                onOpen={handleDrawerOpen}
                onClose={handleDrawerClose}
                variant={drawerVariant}
                className={classes.drawer}
                classes={{ paper: classes.drawerPaper }}
            />
            <Box mt={2} />
            <main className={classes.content}>
                <Toolbar />
                {children}
            </main>
        </div>
    )
}

export default Layout
