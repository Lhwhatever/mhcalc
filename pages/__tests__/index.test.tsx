import React from 'react'
import IndexPage from '../index'
import { createShallow } from '@material-ui/core/test-utils'
import Layout from '../../components/layout/Layout'

describe('/ tests', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    it('should use Layout', () => {
        const wrapper = shallow(<IndexPage />)
        expect(wrapper.find(Layout)).toHaveLength(1)
    })
})
