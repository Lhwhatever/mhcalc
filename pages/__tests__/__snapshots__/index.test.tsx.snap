// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ tests should render correctly 1`] = `
<Layout>
  <WithStyles(ForwardRef(Typography))
    variant="h3"
  >
    Welcome!
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    paragraph={true}
    variant="body1"
  >
    This site hosts several different calculators for MouseHunt to aid in optimizing your hunting.
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    paragraph={true}
    variant="body2"
  >
    MouseHunt is a trademark of HitGrab Inc. I do not claim ownership of MouseHunt, its associated trademarks or its art assets.
  </WithStyles(ForwardRef(Typography))>
  <Component />
</Layout>
`;

exports[`mounted IndexButtons should render correctly 1`] = `
<Component>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    flexDirection="column"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <Styled(AsNextLink(WithStyles(ForwardRef(Button))))
        href="/vrift"
        key="/vrift"
        variant="outlined"
        width="100%"
      >
        <AsNextLink(WithStyles(ForwardRef(Button)))
          className="AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
          href="/vrift"
          variant="outlined"
          width="100%"
        >
          <Link
            href="/vrift"
            passHref={true}
          >
            <WithStyles(ForwardRef(Button))
              className="AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
              href="/vrift"
              onClick={[Function]}
              onMouseEnter={[Function]}
              variant="outlined"
              width="100%"
            >
              <ForwardRef(Button)
                className="AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                href="/vrift"
                onClick={[Function]}
                onMouseEnter={[Function]}
                variant="outlined"
                width="100%"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-outlined AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  href="/vrift"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  type="button"
                  width="100%"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-outlined AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    href="/vrift"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    type="button"
                    width="100%"
                  >
                    <a
                      aria-disabled={false}
                      className="MuiButtonBase-root MuiButton-root MuiButton-outlined AsNextLink(WithStyles(ForwardRef(Button)))-root-4"
                      href="/vrift"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      width="100%"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Valour Rift Simulator
                      </span>
                      <WithStyles(memo)
                        center={false}
                      >
                        <ForwardRef(TouchRipple)
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </a>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </Link>
        </AsNextLink(WithStyles(ForwardRef(Button)))>
      </Styled(AsNextLink(WithStyles(ForwardRef(Button))))>
    </div>
  </Styled(MuiBox)>
</Component>
`;
